packages/belos/src/BelosPseudoBlockGmresSolMgr.hpp

```diff
@@ -1167,11 +1229,55 @@ PseudoBlockGmresSolMgr<ScalarType,MV,OP>::getValidParameters() const
       "The type of scaling used in the explicit residual convergence test.");
     pl->set("Timer Label", label_default_,
       "The string to use as a prefix for the timer labels.");
-    pl->set("Orthogonalization", orthoType_default_,
+
+    //-----------------------------------------------------------------------
+    // Gather the orthoManager's default parameters
+
+    // DGKS ortho parameter list
+    {
+      typedef DGKSOrthoManager<ScalarType, MV, OP> ortho_type;
+
+      RCP<ParameterList> orthoParams = sublist(pl, "DGKS");
+      orthoParams->setParameters( *(ortho_type::getDefaultParameters()) );
+      // override the depTol using this class's defined value (legacy value)
+      orthoParams->set("depTol", orthoKappa_default_);
+    }
+    // ICGS
+    {
+      typedef ICGSOrthoManager<ScalarType, MV, OP> ortho_type;
+
+      RCP<ParameterList> orthoParams = sublist(pl, "ICGS");
+      orthoParams->setParameters( *(ortho_type::getDefaultParameters()) );
+    }
+    // IMGS
+    {
+      typedef IMGSOrthoManager<ScalarType, MV, OP> ortho_type;
+
+      RCP<ParameterList> orthoParams = sublist(pl, "IMGS");
+
+      orthoParams->setParameters( *(ortho_type::getDefaultParameters()) );
+    }
+    #ifdef HAVE_BELOS_TSQR
+    // TSQR
+    {
+      typedef TsqrMatOrthoManager<ScalarType, MV, OP> ortho_type;
+      RCP<ortho_type> ortho_tmp = rcp (new ortho_type ());
+
+      RCP<ParameterList> orthoParams = sublist(pl, "TSQR");
+      orthoParams->setParameters( *(ortho_tmp->getValidParameters()) );
+    }
+    #endif // HAVE_BELOS_TSQR
+    //-----------------------------------------------------------------------
+
+    // legacy parameters
+    Teuchos::RCP<Teuchos::ParameterList> orthoPl = Teuchos::sublist(pl, "Orthogonalziation");
```

Comment:
This is cruft. And why you saw that odd output.  I hadn't generated that list of 'valid parameters' before I emailed them.  This was from a first shot at this. The motivation  was to have a single sublist for the ortho parameters ... but that is foolish, each Ortho kernel has its own set of parameters. It also requires a 'Name' parameter somewhere, e.g., added to each OrthoManager's valdParameters, that way you know which orthogonalization to instantiated.

Instead, I deleted *most* of that... and went with a more natural OO approach.
The solver keeps the original Orthogonalization string (DGKS, ...), and that is used to look up an optional sublist. This lets you append all valid ortho manager paramlists as sublists using their default names (ICGS, DGKS, etc..). Thus creating a valid parameter list in the solver that is both comprehensive and allows validation.

I'll need to clean up a few more things as well, but I wanted to get criticism on the approach before I spend more time on it (pushing to other solvers).